temporal-sdk/src/main/java/io/temporal/internal/sync/SyncWorkflowWorker.java

```diff
@@ -146,24 +176,39 @@ public boolean isStarted() {
 
   @Override
   public boolean isShutdown() {
-    return workflowWorker.isShutdown() && laWorker.isShutdown();
+    return workflowWorker.isShutdown() || laWorker.isShutdown() || heartbeatExecutor.isShutdown();
   }
 
   @Override
   public boolean isTerminated() {
-    return workflowWorker.isTerminated() && laWorker.isTerminated();
+    return workflowWorker.isTerminated()
+        && laWorker.isTerminated()
+        && heartbeatExecutor.isTerminated();
   }
 
   @Override
-  public void shutdown() {
-    laWorker.shutdown();
-    workflowWorker.shutdown();
+  public CompletableFuture<Void> shutdown(ShutdownManager shutdownManager) {
+    return workflowWorker
+        .shutdown(shutdownManager)
+        // we want to shutdown heartbeatExecutor before activity worker, so in-flight activities
+        // could get an ActivityWorkerShutdownException from their heartbeat
+        .thenCompose(
+            ignore ->
+                shutdownManager.shutdownExecutor(
+                    heartbeatExecutor, this + "#heartbeatExecutor", Duration.ofSeconds(5)))
+        .thenCompose(ignore -> laWorker.shutdown(shutdownManager))
+        .exceptionally(
```

Comment:
Log and swallow.
The idea is that shutdown shouldn't throw. If it does - let's ignore it and still allow other steps to perform.
Ideally, we should never end up there. All steps should not be throwing. I added it just in case something breaks this contract or something goes wrong.